### PR TITLE
SAAS-4128: Improve workspace load time by increasing file checking concurrency

### PR DESCRIPTION
**Pull request by ori-moisis**
Before this change each NaCl file would be checked separately, since checking
involves both IO (reading the file) and compute (comparing md5) this was inefficient.

increasing concurrency will improve workspace load times for workspaces with many NaCl files

---

---
_Release Notes_: 
Core:
- Improve workspace load time for workspaces with a large number of NaCl files

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts [hunk 1]
```diff
@@ -631,9 +639,9 @@ const buildNaclFilesSource = (
     ) ?? []
     return (await Promise.all(topLevelFiles.map(async filename => {
       const fragments = await (await (await getState()).parsedNaclFiles.get(filename)).elements()
-      return values.isDefined(
-        await awu(fragments ?? []).find(fragment => resolvePath(fragment, elemID) !== undefined)
-      ) ? filename : undefined
+      return wu(fragments ?? []).some(fragment => resolvePath(fragment, elemID) !== undefined)
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:642]: unrelated change, there was just no reason to use `awu` there...

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts [hunk 2]
```diff
@@ -428,7 +429,7 @@ const buildNaclFilesState = async ({
         referencedIndexDeletions[elementFullName].add(oldNaclFile.filename)
       })
       const oldNaclFileElements = await oldNaclFile.elements() ?? []
-      await awu(oldNaclFileElements).forEach(element => {
+      oldNaclFileElements.forEach(element => {
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:432]: unrelated change, there was just no reason to use `awu` here...

**Review by adamro — APPROVED** [review]

File: packages/workspace/src/workspace/nacl_files/nacl_files_source.ts [hunk 3]
```diff
@@ -543,21 +544,28 @@ const buildNaclFilesSource = (
...
+        CACHE_READ_CONCURRENCY,
+      )
+      await withLimitedConcurrency(
+        wu(naclFilenames).map(filename => async () => {
+          if (!fileNames.has(filename)) {
+            modifiedNaclFiles.push(await naclFilesStore.get(filename) ?? { filename, buffer: '' })
+          }
+          fileNames.add(filename)
+        }),
+        CACHE_READ_CONCURRENCY,
```
Comment (adamro) [inline packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:567]: Is this const legit for use here?

Comment (ori-moisis) [reply packages/workspace/src/workspace/nacl_files/nacl_files_source.ts:567]: basically we are reading files and checking the cache.
it was either `CACHE_READ_CONCURRENCY` or `PARSE_CONCURRENCY`, both technically limit the number of files we read concurrently, but I felt that since we check the cache and not parse, it would make more sense to use the cache related constant.

either way all these limits are set to the same value currently (100)
